Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -81,73 +83,90 @@ namespace ROS2
+                        report += "# " + tr("XACRO parsing failed") + "\n";
+                        report += "\n\n## " + tr("Command called") + "\n\n`" + QString::fromUtf8(outcome.m_called.data()) + "`";
+                        report += "\n\n" + tr("Process failed");
+                        report += "\n\n## " + tr("Error output") + "\n\n";
+                        report += "```\n";
+                        if (outcome.m_logErrorOutput.size())
+                        {
+                            report +=
+                                QString::fromUtf8(outcome.m_logErrorOutput.data(), static_cast<int>(outcome.m_logErrorOutput.size()));
+                        }
+                        else
+                        {
+                            report += tr("(EMPTY)");
+                        }
+                        report += "\n```";
+                        report += "\n\n## " + tr("Standard output") + "\n\n";
+                        report += "```\n";
+                        if (outcome.m_logStandardOutput.size())
+                        {
+                            report += QString::fromUtf8(
+                                outcome.m_logStandardOutput.data(), static_cast<int>(outcome.m_logStandardOutput.size()));
+                        }
+                        else
+                        {
+                            report += tr("(EMPTY)");
+                        }
+                        report += "\n```";
+                        m_checkUrdfPage->ReportURDFResult(report, false);
+                        return;
                     }
-                    report += "\n```";
-                    report += "\n\n## " + tr("Standard output") + "\n\n";
-                    report += "```\n";
-                    if (outcome.m_logStandardOutput.size())
-                    {
-                        report += QString::fromLocal8Bit(
-                            outcome.m_logStandardOutput.data(), static_cast<int>(outcome.m_logStandardOutput.size()));
-                    }
-                    else
-                    {
-                        report += tr("(EMPTY)");
-                    }
-                    report += "\n```";
-                    m_checkUrdfPage->ReportURDFResult(report, false);
-                    m_parsedUrdf = nullptr;
-                    return;
                 }
             }
             else if (Utils::IsFileUrdf(m_urdfPath))
             {
                 // standard URDF
-                m_parsedUrdf = UrdfParser::ParseFromFile(m_urdfPath.Native());
+                // Read the SDF Settings from the Settings Registry into a local struct
+                SdfAssetBuilderSettings sdfBuilderSettings;
+                sdfBuilderSettings.LoadSettings();
+                // Set the parser config settings for URDF content
+                sdf::ParserConfig parserConfig;
+                parserConfig.URDFSetPreserveFixedJoint(sdfBuilderSettings.m_urdfPreserveFixedJoints);
```

Comment:
I tend to try to avoid EBus calls these days.
Honestly for this code we just need a utility helper function that can fill out the parser config with values from the Settings Registry.
Perhaps a function such as `sdf::ParserConfig InitParserConfig()`